template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidator.java

```diff
@@ -137,16 +148,25 @@ private void validateRequiredRoleCountInCluster(
                     .filter(entry -> entry.getValue().contains(role))
                     .map(Entry::getKey)
                     .collect(Collectors.toSet());
-            instanceGroups.stream()
+            List<InstanceGroup> filteredInstanceGroups = instanceGroups.stream()
                     .filter(instanceGroup -> instanceGroupsWithRoles.contains(instanceGroup.getGroupName()))
-                    .map(instanceGroup ->
-                            instanceGroup.getNodeCount() + instanceGroupAdjustments.getOrDefault(instanceGroup.getGroupName(), 0))
+                    .collect(Collectors.toList());
+            String groupNames = filteredInstanceGroups.stream().map(InstanceGroup::getGroupName).collect(Collectors.joining(","));
+            LOGGER.debug("Instance groups with role: {}", groupNames);
+            filteredInstanceGroups.stream().map(instanceGroup -> {
+                        int nodeCount = instanceGroup.getNodeCount() + instanceGroupAdjustments.getOrDefault(instanceGroup.getGroupName(), 0);
+                        LOGGER.debug("{} node has in group of {} with role: {}, instance count with the node count: {}", instanceGroup.getNodeCount(),
+                                instanceGroup.getGroupName(), role, nodeCount);
+                        return nodeCount;
+                    })
                     .reduce(Integer::sum)
                     .ifPresent(remainingInstances -> {
+                        LOGGER.debug("{} remaining instances, required: {} in groups of {} for {}", remainingInstances, requiredCount, groupNames, role);
                         if (remainingInstances < requiredCount) {
                             throw new BadRequestException(String.format(
-                                    "Scaling adjustment is not allowed, based on the template it would eliminate all the instances with "
-                                            + "%s role which is not supported.", role));
+                                    "Scaling adjustment is not allowed in groups of %s, based on the template it would eliminate all the instances with "
+                                            + "%s role which is not supported. %s remained instances but %s the required", groupNames, remainingInstances,
+                                    requiredCount, role));
```

Comment:
* `role` this should come right after `groupNames`.
* `%s remained instances but %s the required` -> `%s remaining instances but %s are required`